src/OrbitGl/ButtonTest.cpp

```diff
@@ -75,4 +75,29 @@ TEST(Button, LabelWorksAsExpected) {
   EXPECT_EQ(button.GetLabel(), kLabel);
 }
 
+TEST(Button, MouseReleaseCallback) {
+  orbit_gl::CaptureViewElementTester tester;
+  Button button(nullptr, tester.GetViewport(), tester.GetLayout());
+
+  uint32_t mouse_released_called = 0;
+  auto callback = [&](Button* button_param) {
+    if (button_param == &button) mouse_released_called++;
+  };
+  button.SetMouseReleaseCallback(callback);
+
+  uint32_t mouse_released_called_expected = 0;
+  button.OnRelease();
+  EXPECT_EQ(mouse_released_called, ++mouse_released_called_expected);
+  button.OnRelease();
+  EXPECT_EQ(mouse_released_called, ++mouse_released_called_expected);
+  button.SetMouseReleaseCallback(nullptr);
+  button.OnRelease();
+  EXPECT_EQ(mouse_released_called, mouse_released_called_expected);
+
+  Button button_with_wrong_callback(nullptr, tester.GetViewport(), tester.GetLayout());
```

Comment:
I'm not really sure what real case you want to test here. Could you give me an example? (ideally making a comment as well)

Reply:
I was thinking of verifying that a button would always send itself as parameter, so I added this to make sure the tests above would not "by accident" be false-positive because there's only one button defined in scope. I can add a comment.